### ISSUE TITLE
Unrecognized git state "DA"

### ISSUE THREAD

**Issue by IllustratedMan-code** [nvim-tree/nvim-tree.lua#1904]
### Description

I just ran into this issue and was directed here. When opening nvim-tree, I get the following error:
```
[NvimTree] 'Unrecognized git state "DA". Please open up an issue on https://github.com/nvim-tree/nvim-tree.lua/issues with this message.' 
```

### Neovim version

```text
NVIM v0.8.2
Build type: Release
LuaJIT 2.1.0-beta3
```

### Operating system and version

NixOS (unstable)

### nvim-tree version

b2ba6de

### Minimal config

```lua
Not really sure what to do for this as I use nix, will try to get something together if needed.
```

### Steps to reproduce

I can't seem to reliably reproduce all the time, but when I can here are the steps:
1. open neovim with `nvim`
2. run `:NvimTreeOpen`

**Comment by IllustratedMan-code** [comment]
```On branch main
Your branch is up to date with 'origin/main'.

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	deleted:    flake.lock
	modified:   flake.nix
	modified:   personal/default.nix
	new file:   personal/hardware-configuration.nix

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	new file:   flake.lock
	modified:   users/programs/nvim/default.nix
	modified:   users/users.nix
```

**Comment by gegoune** [comment]
@alex-courtis looks like
- file removed and staged
- unstaged file with the same name readded
How do we want to highlight that?

**Comment by IllustratedMan-code** [comment]
Looks good to me!

**Comment by gegoune** [comment]
Thanks for confirming. Will merge in once master history issue is resolved (#1906).

**Comment by alex-courtis** [comment]
> * How do we want to highlight that?

Dirty works for me. You can only have so many icons/highlights.
